packages/framework-core/test/booster-command-dispatcher.test.ts

```diff
@@ -109,12 +109,56 @@ describe('the `BoosterCommandsDispatcher`', () => {
         requestID: '42',
       }
 
-      await new BoosterCommandDispatcher(config as any, logger).dispatchCommand(commandEnvelope as any)
+      await new BoosterCommandDispatcher(config as any, logger).dispatchCommand(commandEnvelope as any, {} as any)
 
       expect(fakeHandler).to.have.been.calledWithMatch(commandValue)
     })
 
-    it('properly handle the registered events', async () => {
+    it('allows the handler set the responseHeaders', async () => {
+      class ProperlyHandledCommand {
+        public static handle(command: ProperlyHandledCommand, register: Register) {
+          register.responseHeaders['Test-Header'] = 'test'
+        }
+      }
+
+      spy(ProperlyHandledCommand, 'handle')
+      replace(RegisterHandler, 'handle', fake())
+
+      const config = {
+        commandHandlers: {
+          ProperlyHandledCommand: {
+            authorizedRoles: 'all',
+            before: [],
+            class: ProperlyHandledCommand,
+          },
+        },
+        currentVersionFor: fake.returns(1),
+      }
+      const commandValue = {
+        something: 'to handle',
+      }
+
+      const commandEnvelope = {
+        typeName: 'ProperlyHandledCommand',
+        version: 'π', // JS doesn't care, and π is a number after all xD...
+        currentUser: {
+          roles: ['Loki'],
+        },
+        value: commandValue,
+        requestID: '42',
+      }
+
+      const context = {
+        responseHeaders: {},
+      }
+
+      await new BoosterCommandDispatcher(config as any, logger).dispatchCommand(commandEnvelope as any, context as any)
+
+      expect(ProperlyHandledCommand.handle).to.have.been.calledWithMatch(commandValue, { responseHeaders: {} })
+      expect(context.responseHeaders).to.deep.equal({ 'Test-Header': 'test' })
+    })
```

Comment:
Added this test that checks that the handler can change the `responseHeaders` in the context.